compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -1356,32 +1356,46 @@ impl<'hir> LoweringContext<'_, 'hir> {
         // keep track of the Span info. Now, `add_implicitly_sized` in `AstConv` checks both param bounds and
         // where clauses for `?Sized`.
         for pred in &generics.where_clause.predicates {
-            if let WherePredicate::BoundPredicate(ref bound_pred) = *pred {
-                'next_bound: for bound in &bound_pred.bounds {
-                    if let GenericBound::Trait(_, TraitBoundModifier::Maybe) = *bound {
-                        // Check if the where clause type is a plain type parameter.
-                        match self
-                            .resolver
-                            .get_partial_res(bound_pred.bounded_ty.id)
-                            .map(|d| (d.base_res(), d.unresolved_segments()))
+            let bound_pred = match *pred {
+                WherePredicate::BoundPredicate(ref bound_pred) => bound_pred,
+                _ => continue,
+            };
+            let mut is_param: Option<bool> = None;
+            for bound in &bound_pred.bounds {
+                if !matches!(*bound, GenericBound::Trait(_, TraitBoundModifier::Maybe)) {
+                    continue;
+                }
+                // We only need to compute this once per `WherePredicate`, but don't
+                // need to compute this at all unless there is a Maybe bound.
+                // This closure should be able to be moved out of the loop,
+                // but `get_partial_res` takes a `&mut self`
```

Comment:
`get_partial_res` should take `&self`, it does not need to be mutable.

Reply:
Yeah, I wasn't sure if it was worth changing that for; I wasn't sure if there's some design reason that it's `&mut self`. I'll go ahead and do that though.